src/test/java/io/zold/api/WalletsInTest.java

```diff
@@ -81,19 +81,30 @@ public void createsWalletInFolder() throws IOException {
         );
     }
 
-    // @todo #12:30min Ensure creating a new wallet in a path where a
-    //  wallet with the same id throws an exception and does not
-    //  overwrite the existing one. This could be done using by
-    //  using a Random object with a similar seed in WalletsIn
-    //  and creating a wallet twice for example. When it is done
-    //  enable the test.
     @Test
-    @Ignore("see todo above")
-    public void doesNotOverwriteExistingWallet() throws IOException {
+    public void doesNotOverwriteExistingWallet() throws Exception {
         final Path path = this.folder.newFolder().toPath();
-        new WalletsIn(path).create();
+        final Random random = new FkRandom();
+        new WalletsIn(path, random).create();
         this.thrown.expect(IOException.class);
-        this.thrown.expectMessage("duplicate");
-        new WalletsIn(path).create();
+        this.thrown.expectMessage("already exists");
+        new WalletsIn(path, random).create();
+    }
+
+    /**
+     * Fake randomizer that returns the same value each time.
+     */
+    private static class FkRandom extends Random {
+
+        /**
+         * Serial version.
+         */
+        private static final long serialVersionUID = 2905348968220129619L;
+
+        @Override
+        public long nextLong() {
+            // @checkstyle MagicNumberCheck (1 line)
+            return 16725L;
```

Comment:
@Vatavuk let's allow ID to be specified in constructor instead of hard-coding it here. It may be useful for future tests.